other/upgrade_2-1_mysql.sql

```diff
@@ -115,6 +115,43 @@ if (@$modSettings['smfVersion'] < '2.1')
 ---}
 ---#
 
+---# Parsing board descriptions and names
+---{
+if (@$modSettings['smfVersion'] < '2.1')
+{
+    $request = $smcFunc['db_query']('', '
+        SELECT name, description, id_board
+        FROM {db_prefix}boards');
+
+    $inserts = array();
+
+    $smcFunc['db_free_result']($request);
+
+    while ($row = $smcFunc['db_fetch_assoc']($request))
+    {
+        $inserts[] = array(
+            'name' => $smcFunc['htmlspecialchars'](strip_tags(html_to_bbc($row['name']))),
+            'description' => $smcFunc['htmlspecialchars'](strip_tags(html_to_bbc($row['description']))),
+            'id' => $row['id'],
+        );
+    }
+
+    if (!empty($inserts))
```

Comment:
The check should be unneed because the array when empty should run zero times in the foreach?

Reply:
Wrong guess,
you could use ``forearch((array)  $inserts as $insert)``